lib/bridge/IrcBridge.js

```diff
@@ -360,11 +361,20 @@ IrcBridge.prototype._addRequestCallbacks = function() {
             logMessage(req, "IGNORE not mapped");
             return; // these aren't true successes so don't skew graphs
         }
+        else if (res === BridgeRequest.ERR_DROPPED) {
+            logMessage(req, "IGNORE dropped");
+            if (this._timers) {
+                let timer = this._timers[
+                    isFromIrc ? "remote_request_seconds" : "matrix_request_seconds"
+                ];
+                timer.observe({outcome: "dropped"}, req.getDuration() / 1000);
```

Comment:
It would be nice to factor out the 4 lines above as they seem to repeat a lot. Or possibly the entire if, but then you'd still need some sort of `if (...) return` .